File: app/src/main/java/pl/piotrskiba/angularowo/fragments/ReportsHistoryFragmentContainerFragment.java
package pl.piotrskiba.angularowo.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.viewpager2.widget.ViewPager2;

import com.google.android.material.tabs.TabLayout;
import com.google.android.material.tabs.TabLayoutMediator;

import butterknife.BindView;
import butterknife.ButterKnife;
import pl.piotrskiba.angularowo.R;
import pl.piotrskiba.angularowo.adapters.ViewPagerFragmentAdapter;

public class ReportsHistoryFragmentContainerFragment extends Fragment {

    @BindView(R.id.tablayout)
    TabLayout mTabLayout;

    @BindView(R.id.viewpager)
    ViewPager2 mViewPager;

    public ReportsHistoryFragmentContainerFragment(){

    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_reports_history_fragment_container, container, false);

        ButterKnife.bind(this, view);

        ActionBar actionbar = ((AppCompatActivity)getActivity()).getSupportActionBar();
        actionbar.setTitle(R.string.actionbar_title_report_list);

        ViewPagerFragmentAdapter adapter = new ViewPagerFragmentAdapter(getChildFragmentManager(), getLifecycle());
        adapter.addFragment(new ReportsHistoryFragment(false));
        adapter.addFragment(new ReportsHistoryFragment(true));

        mViewPager.setUserInputEnabled(false);
        mViewPager.setOrientation(ViewPager2.ORIENTATION_HORIZONTAL);
        mViewPager.setAdapter(adapter);

        new TabLayoutMediator(mTabLayout, mViewPager, (tab, position) -> {
            if(position == 0)
                tab.setText(R.string.reports_own);
            else
                tab.setText(R.string.reports_others);
        }).attach();

        return view;
    }
}

File: app/src/main/java/pl/piotrskiba/angularowo/fragments/ReportsHistoryFragmentContainerFragment.kt
package pl.piotrskiba.angularowo.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager2.widget.ViewPager2
import butterknife.BindView
import butterknife.ButterKnife
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayoutMediator
import com.google.android.material.tabs.TabLayoutMediator.TabConfigurationStrategy
import pl.piotrskiba.angularowo.R
import pl.piotrskiba.angularowo.adapters.ViewPagerFragmentAdapter
import pl.piotrskiba.angularowo.fragments.base.BaseFragment

class ReportsHistoryFragmentContainerFragment : BaseFragment() {

    @BindView(R.id.tablayout)
    lateinit var mTabLayout: TabLayout

    @BindView(R.id.viewpager)
    lateinit var mViewPager: ViewPager2

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_reports_history_fragment_container, container, false)

        ButterKnife.bind(this, view)

        val actionbar = (activity as AppCompatActivity?)?.supportActionBar
        actionbar?.setTitle(R.string.actionbar_title_report_list)

        val adapter = ViewPagerFragmentAdapter(childFragmentManager, lifecycle)
        adapter.addFragment(ReportsHistoryFragment.newInstance(false))
        adapter.addFragment(ReportsHistoryFragment.newInstance(true))
        mViewPager.adapter = adapter

        mViewPager.isUserInputEnabled = false
        mViewPager.orientation = ViewPager2.ORIENTATION_HORIZONTAL

        TabLayoutMediator(mTabLayout, mViewPager, TabConfigurationStrategy { tab: TabLayout.Tab, position: Int ->
            if (position == 0)
                tab.setText(R.string.reports_own)
            else
                tab.setText(R.string.reports_others)
        }).attach()

        return view
    }
}